python-avd/pyavd/_eos_designs/structured_config/underlay/loopback_interfaces.py

```diff
@@ -21,84 +23,77 @@ class LoopbackInterfacesMixin(Protocol):
+            description=self.shared_utils.interface_descriptions.router_id_loopback_interface(
                 InterfaceDescriptionData(
                     shared_utils=self.shared_utils,
                     interface="Loopback0",
                     description=default(self.inputs.overlay_loopback_description, self.inputs.router_id_loopback_description),
                 ),
             ),
-            "shutdown": False,
-            "ip_address": f"{self.shared_utils.router_id}/32",
-        }
+            shutdown=False,
+            ip_address=f"{self.shared_utils.router_id}/32",
+        )
 
         if self.shared_utils.ipv6_router_id is not None:
-            loopback0["ipv6_address"] = f"{self.shared_utils.ipv6_router_id}/128"
+            loopback0.ipv6_address = f"{self.shared_utils.ipv6_router_id}/128"
 
         if self.shared_utils.underlay_ospf:
-            loopback0["ospf_area"] = self.inputs.underlay_ospf_area
+            loopback0.ospf_area = self.inputs.underlay_ospf_area
 
         if self.shared_utils.underlay_ldp:
-            loopback0["mpls"] = {"ldp": {"interface": True}}
+            loopback0.mpls.ldp.interface = True
 
         if self.shared_utils.underlay_isis:
-            isis_config = {"isis_enable": self.shared_utils.isis_instance_name, "isis_passive": True}
+            loopback0._update(isis_enable=self.shared_utils.isis_instance_name, isis_passive=True)
             if self.shared_utils.underlay_sr:
-                isis_config["node_segment"] = {"ipv4_index": self._node_sid}
+                loopback0.node_segment.ipv4_index = self._node_sid
                 if self.shared_utils.underlay_ipv6:
-                    isis_config["node_segment"].update({"ipv6_index": self._node_sid})
-
-            loopback0.update(isis_config)
-
-        loopback0 = {key: value for key, value in loopback0.items() if value is not None}
+                    loopback0.node_segment.ipv6_index = self._node_sid
 
-        loopback_interfaces.append(loopback0)
+        self.structured_config.loopback_interfaces.append(loopback0)
 
         # VTEP loopback
         if (
             self.shared_utils.overlay_vtep is True
             and self.shared_utils.vtep_loopback.lower() != "loopback0"
             and self.shared_utils.vtep_loopback.lower().startswith("lo")
         ):
-            vtep_loopback = {
-                "name": self.shared_utils.vtep_loopback,
-                "description": self.shared_utils.interface_descriptions.vtep_loopback_interface(
+            vtep_loopback = EosCliConfigGen.LoopbackInterfacesItem(
+                name=self.shared_utils.vtep_loopback,
+                description=self.shared_utils.interface_descriptions.vtep_loopback_interface(
                     InterfaceDescriptionData(
                         shared_utils=self.shared_utils, interface=self.shared_utils.vtep_loopback, description=self.inputs.vtep_loopback_description
                     )
                 ),
```

Comment:
In some corner cases we can end up with an empty description. In that case EOS will fail. So here we should remember an `or None` after.
```suggestion
                description=self.shared_utils.interface_descriptions.vtep_loopback_interface(
                    InterfaceDescriptionData(
                        shared_utils=self.shared_utils, interface=self.shared_utils.vtep_loopback, description=self.inputs.vtep_loopback_description
                    )
                ) or None,
```

Reply:
Fixed!